api/src/main/java/org/openmrs/Order.java

```diff
@@ -32,10 +32,18 @@
  * 
  * @version 1.0
  */
-public class Order extends BaseOpenmrsData implements java.io.Serializable {
+public class Order extends BaseOpenmrsData implements java.io.Serializable, Comparable{
 	
 	public static final long serialVersionUID = 4334343L;
-	
+
+	@Override
+	public int compareTo(Object o) {
```

Comment:
- You don't need the second if clause because compareWithNullAsGreatest already takes care of that.
-  I would also say that if the passed in order object is null, i guess it gets pushed to the end, i would return maximum integer instead of 0